frontend/src/components/MapLegend.tsx

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+
+interface LegendProps {
+  modeColors: { [key: string]: string };
+}
+
+const MapLegend: React.FC<LegendProps> = ({ modeColors }) => {
+  return (
+    <div className="flex flex-col">
+      <div className="font-bold text-md mb-2">Legend</div>
+      <div className="flex flex-wrap">
+        {Object.entries(modeColors).map(([mode, color]) => (
+          <div key={mode} className="flex items-center mb-2 mr-4">
+            <div
+              className="w-4 h-4 mr-2 rounded-full"
+              style={{ backgroundColor: color }}
+            ></div>
+            <div>{mode}</div>
```

Comment:
can you add class to make the text smaller on mobile. rn it spans two lines but if it's smaller it'll probably look better. so something like `md:text-base text-small` or however it's done with tailwind. Not sure if it's md or lg so just try both and see if it looks fine.

Reply:
made text "md:text-base text-[10px]", it fits most mobile screens. Still doesn't work for phone dimensions under 400 px width